File: carebridge-api/src/main/java/com/carebridge/carebridge_api/core/responses/ErrorResponse.java
package com.carebridge.carebridge_api.core.responses;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class ErrorResponse<T> {
    private String status;
    private String message;
    private List<ErrorDetails> errors;
    private LocalDateTime timestamp;

    public ErrorResponse(String status, String message, List<ErrorDetails> errors) {
        this.status = status;
        this.message = message;
        this.errors = errors;
        this.timestamp = LocalDateTime.now();
    }
}

File: carebridge-admin-nextjs/src/types/api/error-response.ts
export interface ErrorDetails {
    field: string;
    message: string;
}

export interface ServerErrorResponse {
    timestamp: string;
    status: string | number;
    message: string;
    errors: ErrorDetails[];
}

export interface ErrorResponse {
    status: number;
    headers: any;
    name: string;
    error: ServerErrorResponse;
    url: string;
}
